src/herder/PendingEnvelopes.h

```diff
@@ -125,7 +125,7 @@ class PendingEnvelopes
 
     void envelopeReady(SCPEnvelope const& envelope);
 
-    bool pop(uint64 slotIndex, SCPEnvelope& ret);
+    bool pop(uint64 minSlotIndex, uint64 maxSlotIndex, SCPEnvelope& ret);
```

Comment:
It would be good to put a comment here about what _exactly_ pop is expected to do. In particular with respect to range inclusivity and preference. It pops an envelope but is this the lowest or highest valued, in or outside of the range endpoints provided?